gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +236,48 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine. Follow this doc:
+  // https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#.
+  private boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    Process process;
+    String result = "";
+    String tmp = null;
+    if ("Linux".equals(osName)) {
```

Comment:
Probably worth splitting into separate Linux and Windows methods

Reply:
Done.

Reply:
You might need to push. I don't see these changes. Also, please don't resolve the conversations.